### DISCUSSION TITLE
KeePassXC MultiDevice/MultiUser

### DISCUSSION THREAD

**Discussion by JavierRodger** [keepassxreboot/keepassxc#9526] [Q&A] [answered]
Hey,

we are using KeePass2 since years with Windows and SMB Share. We never had an issue with DB is out of sync. If someone modify something in the background and try to save, KeePass2 recognize this and ask for sync.

We try to move to KeePassXC (OSX) and Nextcloud (which sync the files from server to local laptop). We startet with a small group of two people. But even that cause 2 times problems with not synced DBs. Maybe this was caused through Nextcloud maybe not. We have to test SMB in the next days.

Anyway, today we tried the KeePassShare feature, sadly this caused immidialtey massive problems. Sync does not work and it crashed the whole DB from one of the test user and he lost hundereds of password. We had to restore KeePass DB from the backup, hopefully we got everything back.

My question in general. It is possible to share a DB through an online SMB connection? E.g. User1 open the file from SMB, User2 open the same file. User1 modify something and User2 will get this info e.g through "sync" manually or reopen, like with KeePass2.

Thanks for help

**Comment by droidmonkey** [comment]
Yes it is possible, we try to detect changes to the underlying database file and merge those into the unlocked database running in your instance. 

In general, however, keepass and keepassxc are not meant to be used on the manner you describe. We are an individual password manager. It just so happens you can share the database file between users and we try to accommodate that behavior.

**Reply by JavierRodger** [reply]
Hi droidmonkey,

thanks for the fast answer.

This sounds great. We may had a problem run into an Nextcloud-Sync issue. We will try with an online file share.
Is there any option, to trigger the sync manually or to get an info, that a sync was done/started?

KeePass2 official support this kind of configuration: https://keepass.info/help/base/multiuser.html

Best Regards
Javier

**Reply by droidmonkey** [reply]
We do the same thing they write there. 

You can't trigger a sync but you can do a database merge with the same file you have open.

**Reply by JavierRodger** [reply]
Hey, ok great news :-)

What would happen if a conflict will happen? e.g. Two user modify at the same time?

**Reply by droidmonkey** [reply, marked as answer]
The entry with the newest modified date wins. The original entry is moved into history. This is the same behavior as keepass.
